File: SDK Project/app/src/main/java/com/cardgame/sdk/data/RemoteDataProvider.java
package com.cardgame.sdk.data;

import android.util.Log;
import com.google.gson.Gson;
import com.cardgame.sdk.HttpUtils;

public class RemoteDataProvider implements IDataProvider {
    @Override
    public PlayerData GetPlayer(int deckSize, boolean isEnemy)
    {
        String url = DataController.URL + "/player/" + deckSize + "/" + isEnemy;

        try {

            String response = HttpUtils.sendGetRequest(url);
            Gson gson = new Gson();

            return gson.fromJson(response, PlayerData.class);
        }
        catch (Exception e)
        {
            Log.e("[SDK]", e.getMessage());
        }

        return null;
    }

    @Override
    public void IncreaseCardPopularity(int dataBaseID) {
        try {

            String url = DataController.URL + "/player/increasePopularity/" + dataBaseID;
            HttpUtils.sendGetRequest(url);
        }
        catch (Exception e)
        {
            Log.e("[SDK]", e.getMessage());
        }
    }
}

File: Unity Project/Assets/Scripts/Data/RemoteDataProvider.cs
using UnityEngine;
using CardGame.Natives;

namespace CardGame.Data
{
    /// <summary>
    /// Handles SDK Requests
    /// </summary>
    public class RemoteDataProvider : IDataProvider
    {
        public RemoteDataProvider(bool simulateServer) => AndroidWrapper.NativeDataController.Init(simulateServer, URLBuilder.GetUrl());

        public PlayerData GetPlayer(int deckSize, bool isEnemy)
        {
            var playerDataJson = AndroidWrapper.NativeDataController.GetRawPlayerData(isEnemy);
            return string.IsNullOrEmpty(playerDataJson) ? null : JsonUtility.FromJson<PlayerData>(playerDataJson);
        }

        public void IncreaseCardPopularity(int dataBaseID) => AndroidWrapper.NativeDataController.IncreaseCardPopularity(dataBaseID);
    }
}
